### ISSUE TITLE
We need a button to be able to disaggregate dam, hospitals, waste water treatment plants  layers so that people have the option of seeing them at different resolutions

### ISSUE THREAD

**Issue by sasignell** [caryfitzhugh/NYCCSC-map-viewer#166]

**Comment by caryfitzhugh** [comment]
How would you split them up further? A drop-down in the layer to select which sections to show?
And how would the be split?

**Comment by colinbeier** [comment]
All they want is a toggle to be able to turn on / off the algorithm that dynamically clumps / separates the points when you zoom the map around.  In other words, to go from points with numbers in them to just seeing all of the points regardless of how it looks in the map (which will be a mess).  Is this possible?  I've told them multiple times that this could be problematic for the user, and would add another layer of complexity to the UI.  If we do this, then we will need to add at least another button / icon to the list for the given layer, and provide a wizard slide explaining what this does.

**Comment by sasignell** [comment]
Maybe could just have one toggle that turns clustering of and on for the
whole map (all layers) rather than individual toggles for each layer.

On Tue, Mar 22, 2016 at 10:31 AM, colinbeier <notifications@github.com>
wrote:

> All they want is a toggle to be able to turn on / off the algorithm that
> dynamically clumps / separates the points when you zoom the map around. In
> other words, to go from points with numbers in them to just seeing all of
> the points regardless of how it looks in the map (which will be a mess). Is
> this possible? I've told them multiple times that this could be problematic
> for the user, and would add another layer of complexity to the UI. If we do
> this, then we will need to add at least another button / icon to the list
> for the given layer, and provide a wizard slide explaining what this does.
>
> On Mar 22, 2016, at 10:23 AM, Cary FitzHugh wrote:
>
> > How would you split them up further? A drop-down in the layer to select
> which sections to show?
> > And how would the be split?
> >
> > —
> > You are receiving this because you are subscribed to this thread.
>
> > Reply to this email directly or view it on GitHub
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/caryfitzhugh/NYCCSC-map-viewer/issues/166#issuecomment-199840116>
>



-- 
Steve Signell
Frontier Spatial, L.L.C.
1062 Dean St.
Schenectady, NY 12309
518-232-1008
frontierspatial.com
steve@frontierspatial.com

**Comment by colinbeier** [comment]
That would work - it would satisfy their request.  They said a button, not many buttons!  :)

**Comment by caryfitzhugh** [comment]
in terms of coding mess, doing it individually is much nicer...

But - one reason to do the clumping is the browser can bog when there are jillions of points.

We could twiddle the clumping settings to do it more granularly.. ?

**Comment by caryfitzhugh** [comment]
I'd like to try to fiddle the settings, and then say "no" if that isn't good enough?

**Comment by colinbeier** [comment]
Sure, sounds good. It is a minor quibble.  I think once they see what it would look / behave like without the clustering, they will miss the clustering.

**Comment by caryfitzhugh** [comment]
@colinbeier on gh-pages... it's more clusters!

**Comment by colinbeier** [comment]
Not sure what I'm looking for here... is there a button to dis-aggregate?

**Comment by caryfitzhugh** [comment]
no - I just fiddled the cluster algorithm.  They cluster at 40px radius, not 80... so it looks like more stuff...

**Comment by colinbeier** [comment]
that's cool.  but i think they wanted the cluster algorithm to be something the user could toggle on / off.  is that possible?

**Comment by caryfitzhugh** [comment]
Sorry - I mean, it's "possible" - but it's basically a re-write of the
renderer.  Clustering is a low-level behavior, so...  You can't just turn
clustiner on / off easily.  It seemed like that was too big an ask for it,
so i tried a compromise... ?

**Comment by colinbeier** [comment]
OK if it's a lot of work I'll let them know, what you did might suffice... thanks!

Colin Beier
SUNY ESF

On Mar 25, 2016 11:31 AM, Cary FitzHugh <notifications@github.com> wrote:
Sorry - I mean, it's "possible" - but it's basically a re-write of the
renderer. Clustering is a low-level behavior, so... You can't just turn
clustiner on / off easily. It seemed like that was too big an ask for it,
so i tried a compromise... ?

**Comment by caryfitzhugh** [comment]
@colinbeier I just fiddled this:
![image](https://cloud.githubusercontent.com/assets/445848/14089465/07b0bc42-f503-11e5-82e8-2a081971a791.png)


That is what the map looks like with no clustering of dams.
It locked my browser for nearly 10 - 15s drawing all of them.
It makes the browser just gag.  Too much.  Too much cap'n!

It's a non-starter. So I'll close this with a commit which does the following:

What I did do, was set the clusters smaller, so it's only a 40px radius, not an 80px radius.
And clustering turns off at zoom level 10 now.
